src/frontend/admin/src/components/templates/enrollments/inputs/RHFSelectEnrollmentState.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from "react";
+import { useMemo } from "react";
+import { defineMessages, useIntl } from "react-intl";
+import {
+  RHFSelect,
+  RHFSelectProps,
+  SelectOption,
+} from "@/components/presentational/hook-form/RHFSelect";
+import { EnrollmentState } from "@/services/api/models/Enrollment";
+import { enrollmentStateMessages } from "@/translations/enrollments/enrollment-state";
+
+const messages = defineMessages({
+  state: {
+    id: "components.templates.enrollments.input.RHFSelectEnrollmentState.state",
+    description: "Label text for the select state input",
+    defaultMessage: "State",
+  },
+});
+
+export function RHFSelectEnrollmentState(props: RHFSelectProps) {
+  const intl = useIntl();
+  const options: SelectOption[] = useMemo(() => {
+    const result: SelectOption[] = [];
+    Object.values(EnrollmentState).forEach((v) => {
+      if (v) {
+        result.push({
+          label: intl.formatMessage(
+            enrollmentStateMessages[v as EnrollmentState],
+          ),
+          value: v + "",
+        });
+      }
+    });
+    return result;
+  }, []);
```

Comment:
You are iterating on `EnrollmentState` enum, so I do not understand why you have to check if `v` is defined ? Furthermore `as EnrollmentState` is useless.

```suggestion
  const options: SelectOption[] = useMemo(() => {
    return Object.values(EnrollmentState).map((v) => ({
      label: intl.formatMessage(enrollmentStateMessages[v]),
      value: v,
    }));
  }, []);
```